Move Mutex.new to constructor

olleolleolle opened the pull request:
This PR moves the `Mutex` creation to the constructor of this class.

  - You could end up with more than one mutex on platforms wihtout a GIL/GVL

Hope this helps!

olleolleolle commented:
cc @ioquatix

ioquatix commented on lib/carrot/amqp/server.rb:
```diff
@@ -125,7 +126,7 @@ def socket
       begin
         # Attempt to connect.
-        mutex.lock if multithread?
+        @mutex.lock if multithread?
```
@olleolleolle you could probably just use `@mutex&.lock`/`@mutex&.unlock`?

olleolleolle replied:
That would be less work for the computer! So, great!

olleolleolle committed:
server: Move Mutex.new to constructor

  - Make the check for multi-processing "Do we have a Mutex?" by using
  safe navigation
  - We do this change because we could end up with more than one Mutex
  on platforms without a GIL/GVL, such as JRuby
